app/src/test/java/org/mozilla/fenix/components/metrics/AppAllSourceStartTelemetryTest.kt

```diff
@@ -0,0 +1,73 @@
+package org.mozilla.fenix.components.metrics
+
+import android.content.Intent
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.verify
+import mozilla.components.support.utils.toSafeIntent
+import org.junit.Before
+import org.junit.Test
+
+internal class AppAllSourceStartTelemetryTest {
```

Comment:
The comment you're looking for about any missing tests. :) Do you know how to run code coverage tools? They're really helpful in identifying lines that weren't run during the tests. After running it, I see:
![image](https://user-images.githubusercontent.com/759372/88327868-6d7cfa80-ccdc-11ea-816e-0063e85f5b5b.png)

The `onApplicationOnStop` method was never called (note the red bars to the left!). That means we never tested that functionality.

From a theory perspective, I'm also concerned that this isn't tested: the boolean logic is the most tricky thing in here so we should **write a test to ensure resetting works as expected.**

**Additionally, there's one more test I'd want to write from a theory perspective: `UNKNOWN`.** This shows a limitation in code coverage tools: since they only show lines you've executed during the test, not what's actually tested, one test passes in UNKNOWN so that line is executed even if we never assert for it.

So please write those two tests and try out the code coverage tools if you haven't done so already! :) It's the shield button in AS, next to the debug button.

If we wanted to be really tested, we could also test what happens if `onApplicationOnStop` is called before an intent is received... but that's probably overkill. :)

Reply:
Thanks! @mcomella  Ive updated the test cases based on this comment